src/machines/java/com/enderio/machines/common/blockentity/base/VacuumMachineBlockEntity.java

```diff
@@ -144,9 +145,9 @@ public void onLoad() {
     }
 
     private void generateParticle(RangeParticleData data, Vec3 pos) {
```

Comment:
this generateParticle method should be a base method (together with the range values).

Reply:
could you expland?

Reply:
Currently, you can find the same method in all classes that have a range (vacuum blocks and powered spawner). It would be better to place it in the base machine class with a getRange method (and storing of that range in nbt) that these classes can overwrite. As to not reimplement it every time.

Reply:
I'm going to say that that's a future PR thing; this PR's scope is already pretty wide